PC-1122 Downloads

ChrisAtPolicyCo opened the pull request.

mccool committed:
add dist directory

mccool committed:
Add util function to return the buffer from Packer

mccool committed:
Export Options type

ChrisAtPolicyCo committed:
PC-1122 add title / sub title, format footer

- added table to footer to format title sub title and page number
- page of total pages in page numbering
- added title to and sub title to table of contents page
- put page break in after table of contents

mccool committed:
Remove borders on footer and auto size footer

ChrisAtPolicyCo committed:
PC-1122 Fix table error, first cell

- clear previous opts when starting a table.

ChrisAtPolicyCo committed:
PC-1122 clear nextRunOpts in closeBlock

- this value was carrying over to the next node, causing the hard_break
  to carry over to the next paragraph.

ChrisAtPolicyCo committed:
PC-1122 added CSS support

ChrisAtPolicyCo committed:
PC-1122 pass paragraph justification left / right / center

ChrisAtPolicyCo committed:
PC-1122 adding code block

ChrisAtPolicyCo committed:
PC-1122 adding margins, default font size hard coded

ChrisAtPolicyCo committed:
PC-1122 fix default font size, default justify left, margin for Controls

dlee3 commented on src/schema.ts:
```diff
@@ -20,14 +21,15 @@ export const defaultNodes: NodeSerializer = {
       HeadingLevel.HEADING_5,
       HeadingLevel.HEADING_6,
     ][node.attrs.level - 1];
+    // TODO pass margin in Header node
```
Does this need to be done before merging?

ChrisAtPolicyCo replied:
no, this was fixed properly.  That was a work around until we Tyler helped add a margin at the end of an article.

dlee3 commented on src/serializer.ts:
```diff
@@ -88,6 +93,8 @@ export class DocxSerializerState<S extends Schema = any> {
     this.children = [];
     this.numbering = [];
+
+    this.options.fontSize = 17;
```
Should we have a `DEFAULT_FONT_SIZE` const somewhere? I noticed you defaulted to 17 in `cssToDocxStyle.ts ` as well.

ChrisAtPolicyCo replied:
yes, I'll make one.

ChrisAtPolicyCo committed:
PC-1122 cleanup comments

ChrisAtPolicyCo committed:
PC-1122 cleanup default value

ChrisAtPolicyCo committed:
PC-1122 fix center justify

ChrisAtPolicyCo committed:
PC-828 Add support for Org Logo

ChrisAtPolicyCo committed:
PC-1337 docx not accepting policy configuration items

-pay attention to, paragraph alignment, font-style, italics, and text-decoration, underline
